FM-Net/fm-net.c

```diff
@@ -169,48 +217,63 @@ void rewrite(Net* net, u32 a_addr) {
 
     // UnaryOperation
     if (a_type == OP1) {
-      u64 dst = enter_port(net, Pointer(a_addr, 2));
-      u32 fst = numb_of(b_ptrn);
-      u32 snd = numb_of(enter_port(net, Pointer(a_addr, 1)));
-      u64 res;
+      union {
+        uint32_t i;
+        float f;
+      } fst, snd, res;
+      u64 dst;
+
+      dst = enter_port(net, Pointer(a_addr, 2));
+      fst.i = numb_of(b_ptrn);
+      snd.i = numb_of(enter_port(net, Pointer(a_addr, 1)));
       switch (a_kind) {
-        case  0: res = Numeric(fst + snd); break;
-        case  1: res = Numeric(fst - snd); break;
-        case  2: res = Numeric(fst * snd); break;
-        case  3: res = Numeric(fst / snd); break;
-        case  4: res = Numeric(fst % snd); break;
-        case  5: res = Numeric((u32)(pow((float)fst, (float)snd))); break;
-        case  6: res = Numeric((u32)(pow((float)fst, ((float)snd / pow(2.0,32.0))))); break;
-        case  7: res = Numeric(fst & snd); break;
-        case  8: res = Numeric(fst | snd); break;
-        case  9: res = Numeric(fst ^ snd); break;
-        case 10: res = Numeric(~snd); break;
-        case 11: res = Numeric(fst >> snd); break;
-        case 12: res = Numeric(fst << snd); break;
-        case 13: res = Numeric(fst > snd ? 1 : 0); break;
-        case 14: res = Numeric(fst < snd ? 1 : 0); break;
-        case 15: res = Numeric(fst == snd ? 1 : 0); break;
-        default: res = 0; printf("[ERROR]\nInvalid interaction."); break;
+      case ADD: res.i = fst.i + snd.i; break;
+      case SUB: res.i = fst.i - snd.i; break;
+      case MUL: res.i = fst.i * snd.i; break;
+      case DIV: res.i = fst.i / snd.i; break;
+      case MOD: res.i = fst.i % snd.i; break;
+      case POW: res.i = powi(fst.i, snd.i); break;
+      case AND: res.i = fst.i & snd.i; break;
```

Comment:
I guess the kind 6 wasn't supposed to be the AND, but rather fst^(snd/2^32), whatever that was supposed to be, right? Don't this break anything? Would be nice to add some regression testing.

Reply:
If you look at the individual commits, they first switch to using the enum (https://github.com/moonad/Formality-C/pull/2/commits/a6ae49f0376248a8f0e4c281579032c4f42efc9c), and then switch to the op codes used in the latest Formality (https://github.com/moonad/Formality-C/pull/2/commits/05fb45454a17bfda84d09d76059ed7ad34d2f87c), which does *not* have that operator.

In the latter commit description, I referred to the corresponding Formality commit, which you can see also removes that operator:
https://github.com/moonad/Formality/commit/94fb510db029ec86ac815b1c57675763ab520f29#diff-5eb486a0616553404e3b1f2431a54e23